webhooky/tests/auth.rs

```diff
@@ -0,0 +1,741 @@
+) -> Result<HttpResponseAccepted<slack_chat_api::InteractivePayload>, HttpError> {
+    let event = body.into_inner()?;
+    let ser_payload = urlencoding::decode(&event.payload).unwrap();
+    let payload: slack_chat_api::InteractivePayload = serde_json::from_str(&ser_payload).unwrap();
+    Ok(HttpResponseAccepted(payload))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/slack/interactive/audit",
+    content_type = "application/x-www-form-urlencoded"
+}]
+async fn hmac_slack_interactive_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBodyAudit<webhooky::handlers_slack::SlackWebhookVerification, InteractiveEvent>,
+) -> Result<HttpResponseAccepted<slack_chat_api::InteractivePayload>, HttpError> {
+    let event = body.into_inner()?;
+    let ser_payload = urlencoding::decode(&event.payload).unwrap();
+    let payload: slack_chat_api::InteractivePayload = serde_json::from_str(&ser_payload).unwrap();
+    Ok(HttpResponseAccepted(payload))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/bearer/verify",
+}]
+async fn bearer_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    _: Bearer<InternalToken>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/bearer/audit",
+}]
+async fn bearer_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    _: BearerAudit<InternalToken>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/token/verify",
+}]
+async fn token_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    _: QueryToken<InternalToken>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/token/audit",
+}]
+async fn token_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    _: QueryTokenAudit<InternalToken>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+static INIT: std::sync::Once = std::sync::Once::new();
+
+/// Setup function that is only run once, even if called multiple times.
+fn setup_logger() {
+    INIT.call_once(|| {
+        pretty_env_logger::init();
+    });
+}
+
+fn make_server() -> (u16, HttpServer<()>) {
+    setup_logger();
+
+    // Configure fake test keys for checking implementations
+    std::env::set_var("INTERNAL_AUTH_BEARER", "TEST_BEARER");
+    std::env::set_var("DOCUSIGN_WH_KEY", "vkPkH4G2k8XNC5HWA6QgZd08v37P8KcVZMjaP4zgGWc=");
+    std::env::set_var("GH_WH_KEY", "vkPkH4G2k8XNC5HWA6QgZd08v37P8KcVZMjaP4zgGWc=");
+    std::env::set_var("SLACK_WH_KEY", "vkPkH4G2k8XNC5HWA6QgZd08v37P8KcVZMjaP4zgGWc=");
+
+    let config_dropshot = ConfigDropshot {
+        bind_address: "127.0.0.1:0".parse().unwrap(),
+        request_body_max_bytes: 107374182400, // 100 Gigiabytes.
```

Comment:
Not that it really matters, but I'd suggest a much smaller max body (maybe 1K?) for the tests.